docs/tekton-bundle-contracts.md

```diff
@@ -0,0 +1,97 @@
+<!--
+---
+linkTitle: "Tekton Bundle Contracts"
+weight: 8
+---
+-->
+# Tekton Bundle Contract v0.1
+
+When using a Tekton Bundle in a task or pipeline reference, the OCI artifact backing the
+bundle must adhere to the following contract.
+
+## Contract
+
+Only Tekton CRDs (eg, `Task` or `Pipeline`) may reside in a Tekton Bundle used as a Tekton
+bundle reference.
+
+Each layer of the image must map 1:1 with a single Tekton resource.
+
+Each layer must contain all of the following annotations:
+
+- `org.opencontainers.image.title` => `ObjectMeta.Name` of the resource
+- `dev.tekton.image.kind` => `TypeMeta.Kind` of the resource, all lowercased (eg, `task`)
+- `dev.tekton.image.apiVersion` => `TypeMeta.APIVersion` of the resource (eg 
+"tekton.dev/v1alpha1")  
+
+The union of the { `dev.tekton.image.apiVersion`, `dev.tekton.image.kind`, `org.opencontainers.image.title` }
+annotations on a given layer must be unique among all layers of that image. In practical terms, this means no two
+"tasks" can have the same name for example.
+
+Each layer must map 1:1 with a single Tekton resource (eg Task), be compressed and stored with a supported OCI MIME type 
+*except* for `+zstd` types. For list of the supported types see
```

Comment:
This is largely an implementation detail of the library you're using and not sure if it should be part of the contract really? It's also my fault this doesn't work yet :)

Reply:
yeah I just mentioned it because Vincent had point out that we were being specific about the MIME type so I figured, so long as we have a caveat in the format of the image, we should mention what we do and don't support?